server/controllers/medical/patient.js

```diff
@@ -429,3 +452,116 @@ function logVisit(patientData, userId) {
 function isEmpty(array) {
   return array.length === 0;
 }
+
+/**
+* GET /patient/search?name={string}&detail={boolean}&limit={number}
+* GET /patient/search?reference={string}&detail={boolean}&limit={number}
+* GET /patient/search?fields={object}
+*
+* @desc This function is responsible to find a patient with detailled informations or not
+* and with a limited rows or not
+*/
+function search (req, res, next) {
+
+  var sql,
+      columns    = null,
+      data       = [],
+      qReference = req.query.reference,
+      qName      = req.query.name,
+      qFields    = req.query.fields,
+      qDetail    = req.query.detail || 0,
+      qLimit     = req.query.limit;
+
+  try {
+    var missingRequiredParameters = (!qReference && !qName && !qFields);
+    if (missingRequiredParameters) { throw req.codes.ERR_PARAMETERS_REQUIRED; }
+
+    qFields = qFields ? JSON.parse(qFields) : null;
+    qDetail = Number(qDetail);
+    qLimit  = Number(qLimit);
+
+  } catch (err) {
+    return next(err);
+  }
+
+  // customize returned columns according detailled results or not
+  if (!qDetail) {
```

Comment:
Can this use a single if (instead of if/else) by doing something like this:
```js
var columns = 
  'q.uuid, q.project_id, q.reference, q.debitor_uuid, ' +
  'q.first_name, q.last_name, q.middle_name, q.sex, q.dob ';

// retrieve more information about the patient on the "?detailed=1" condition.
if (qDetail) {
  columns +=
     'q.father_name, q.mother_name, ' + 
     'q.first_name, q.last_name, q.middle_name, q.sex, q.dob, q.father_name, q.mother_name, ' +
     'q.profession, q.employer, q.spouse, q.spouse_profession, q.spouse_employer, ' +
     'q.religion, q.marital_status, q.phone, q.email, q.address_1, q.address_2, ' +
     'q.renewal, q.origin_location_id, q.current_location_id, q.registration_date, ' +
     'q.title, q.notes, q.hospital_no, q.text, ' +
     'q.account_id, q.price_list_uuid, q.is_convention, q.locked ';
}
```
In this fashion, you do not repeat columns (and it naturally shows that the qDetail condition is an extension of the !qDetail condition).  To me, this option is much cleaner and clearer, but it is a matter of preference.  Let me know what you think, either way works!